concepts/src/concept_csv_export.py

```diff
@@ -266,29 +289,39 @@ def locale_join_snippet(name_types: list, locale: str):
 
         return "\n    ".join(snippets)
 
-    def type_name_transform(mapping_type: str):
-        return mapping_type.replace("-", "_").replace(" ", "_")
+    def name_transform(mapping_type: str):
+        return mapping_type.replace("-", "_").replace(" ", "_").replace(".", "_").replace("|", "_")
 
-    def mapping_select_snippet(mapping_type: str):
-        return "GROUP_CONCAT(DISTINCT term_source_name_{tn}, ':', term_code_{tn} SEPARATOR ';') 'Mappings|{t}'\n".format(
-            t=mapping_type, tn=type_name_transform(mapping_type)
+    def mapping_select_snippet(mapping_type: str, concept_source: str):
+        return "GROUP_CONCAT(DISTINCT term_code_{csn}_{tn} SEPARATOR ';') 'Mappings|{t}|{cs}'\n".format(
+            t=mapping_type, tn=name_transform(mapping_type), cs=concept_source, csn=name_transform(concept_source)
         )
 
-    def mapping_join_snippet(mapping_type: str):
+    def mapping_join_snippet(mapping_type: str, concept_source: str):
+
+        datatype_clause = "";
+
+        if '|' in concept_source:
+            datatype = concept_source.split("|")[1]
+            if datatype == 'Number':
+                datatype_clause = ' WHERE CAST(crt.code as unsigned) != 0'
+            elif datatype == 'Name':
+                datatype_clause = ' WHERE CAST(crt.code as unsigned) = 0'
+
         return (
-            "LEFT JOIN (SELECT crm.concept_id, source.name term_source_name_{tn}, crt.code term_code_{tn} FROM concept_reference_map crm \n"
+            "LEFT JOIN (SELECT crm.concept_id, crt.code term_code_{csn}_{tn} FROM concept_reference_map crm \n"
             "           JOIN concept_map_type map_type ON crm.concept_map_type_id = map_type.concept_map_type_id AND map_type.name = '{t}' \n"
             "           JOIN concept_reference_term crt ON crm.concept_reference_term_id = crt.concept_reference_term_id AND crt.retired = 0 \n"
-            "           JOIN concept_reference_source source ON crt.concept_source_id = source.concept_source_id) term_{tn} \n"
-            "   ON c.concept_id = term_{tn}.concept_id \n"
-        ).format(t=mapping_type, tn=type_name_transform(mapping_type))
+            "           JOIN concept_reference_source source ON crt.concept_source_id = source.concept_source_id and source.name = '{cs}'{dtc}) term_{csn}_{tn} \n"
+            "   ON c.concept_id = term_{csn}_{tn}.concept_id \n"
+        ).format(t=mapping_type, tn=name_transform(mapping_type), cs=concept_source.split("|")[0], csn=name_transform(concept_source), dtc=datatype_clause)
 
     select = (
         "SET SESSION group_concat_max_len = 1000000; "
-        "SELECT c.uuid, MAX(cd.description) 'Description:"
+        "SELECT c.uuid, MAX(REPLACE(REPLACE(cd.description,'\\r',''), '\\n','')) 'Description:"
```

Comment:
Hack to remove newlines and carriage turns in concept decsriptions that were breaking things.